s3auth-relay/src/main/java/com/s3auth/relay/BaseFacade.java

```diff
@@ -0,0 +1,435 @@
+     */
+    private final transient ScheduledExecutorService backend;
+
+    /**
+     * Blocking queue of ready-to-be-processed sockets.
+     */
+    private final transient BlockingQueue<Socket> sockets =
+        new SynchronousQueue<Socket>();
+
+    /**
+     * Server socket.
+     */
+    private final transient ServerSocket server;
+
+    /**
+     * Secured Server socket.
+     */
+    private final transient ServerSocket secured;
+
+    /**
+     * Protected constructor.
+     * @param frontthreads The number of front threads
+     * @param frontprefix The name of the front threads
+     * @param backthreads The number of back threads
+     * @param backprefix The name of the back threads
+     * @param svr The server socket
+     * @param secure The secure server socket
+     * @checkstyle ParameterNumber (5 lines)
+     */
+    protected BaseFacade(final int frontthreads, final String frontprefix,
+        final int backthreads, final String backprefix, final ServerSocket svr,
+        final ServerSocket secure) {
+        this.frontend = this.createThreadPool(frontthreads, frontprefix);
+        this.backend = this.createThreadPool(backthreads, backprefix);
+        this.server = svr;
+        this.secured = secure;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close() throws IOException {
+        try {
+            this.shutdown(this.frontend);
+            this.shutdown(this.backend);
+        } catch (final InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            throw new IOException(ex);
+        } finally {
+            IOUtils.closeQuietly(this.server);
+            IOUtils.closeQuietly(this.secured);
+        }
+    }
+
+    /**
+     * Start listening to the ports.
+     *
+     * The default implementation is <em>empty</em>. This should be overridden
+     * in the subclasses as necessary.
+     */
+    protected void listen() {
+        // nothing to do, intentionally empty
+    }
+
+    /**
+     * Report overflow problem to the socket and close it.
+     *
+     * The default implementation is <em>empty</em>. This should be overridden
+     * in the subclasses as necessary.
+     *
+     * @param socket The socket to report to
+     * @throws IOException If something wrong happens inside
+     */
+    protected void overflow(final Socket socket) throws IOException {
+        // nothing to do, intentionally empty
+    }
+
+    /**
+     * Dispatcher for either HttpThread or FTPThread.
+     * @param thread The dispatcher for processing a thread
+     * @param bckend The executor service
+     */
+    private void threadRunnableDispatcher(final RequestDispatcher thread,
+        final ScheduledExecutorService bckend) {
+        final Runnable runnable = new VerboseRunnable(
+            new BaseFacade.ThreadRunnableDispatcher(thread), true, true
+        );
+        for (int idx = 0; idx < THREADS; ++idx) {
```

Comment:
@simonjenga please add `BaseFacade.` before `THREADS` (all static fields have to be qualified with class name)